File: src/javascript/IterativeMinAndMax.js
const MAX_LENGTH = 10;

/**
 * @param {number[]} vector 
 */
function MinMaxIterative(vector){
    let min = max = vector[0];

    vector.forEach(value => {
        if (value > max) max = value;
        if(value < min) min = value;
    });

    return {
        min, 
        max
    }
}

function main(){
    let vector = [];
    let values = "[";
    for (let index = 0; index < MAX_LENGTH; index++) {
        let value = Math.floor(Math.random() * 99);  // 0 - 99
        vector.push(value);
        values += `${value}, `;
    }
    values += "]"

    console.log(values);
    const result = MinMaxIterative(vector);
    console.log(`Min: ${result.min}\nMax: ${result.max}`);
}

main();

File: src/c/IterativeMinAndMax.c
#include <stdio.h>

int main() {
    int array[8] = { 1, 2, 3, 4, 5, 6, 7, 8 };

    int min = array[0];
    int max = array[0];

    for (int i = 1; i < 8; ++i) {
        if (array[i] < min) {
            min = array[i];
        }
        if (array[i] > max) {
            max = array[i];
        }
    }

    printf("min: %d\n", min);
    printf("max: %d\n", max);
    return 0;
}
